### ISSUE TITLE
Non-IPC constructors for external Arrow data?

### ISSUE THREAD

**Issue by kylebarron** [uwdata/flechette#14]
👋 This is a cool library, and it's nice to see multiple options for Arrow in JS. In particular, I like that you have fuller support for Arrow data, including the new view types. 

I see that right now the only public export is `tableFromIPC`. Are you interested in exporting more constructors (but not builders) for Arrow data?

In particular, I'm excited about zero-copy Arrow interop with WebAssembly: https://github.com/kylebarron/arrow-js-ffi. In the future I want to explore more zero-copy interop with _Python_ running inside [Pyodide](https://pyodide.org/). Many libraries like Polars, DuckDB, etc use string views internally and export data as Arrow string view types, and so being able to represent them is important to me.

But in this case, I'm constructing data from _existing buffers_, so I can't use the `tableFromIPC` function. Are you interested in making something like `Batch` public?

**Comment by jheer** [comment]
Thanks!

- Take a look at https://github.com/uwdata/flechette/pull/13, which makes moves in this direction, including builders and table constructors. (Feedback welcome!)
- I'm also open to exporting Batch (or perhaps more helpfully, the new `batchType` method in #13) if there's a compelling use case!

Maybe take a look and tell me if there's more that you'd need?

**Comment by jheer** [comment]
Update: I merged #13, so you can look at the main branch now as well. I think you might achieve your goals with something like this:

```js
import { Column, batchType, binaryView, tableFromColumns } from '@uwdata/flechette';

const type = binaryView();
const BatchType = batchType(type);
const batch = new BatchType({
  length,    // number of rows in batch
  nullCount, // number of null-valued rows
  type,      // data type
  validity,  // validity bitmap
  values,    // view values buffer
  data       // array of auxiliary view data buffers
});
const table = tableFromColumns({ colName: new Column([batch]) });
```

Of course, this assumes you have prepared the underlying buffers yourself. If you have multiple record batches, simply generate the needed number of batches and pass them to the `Column` constructor.

**Comment by jheer** [comment]
Closing for now. Feel free to re-open (or file another issue) if more is needed.
